src/index.js

```diff
@@ -0,0 +1,91 @@
+/*
+  MIT License http://www.opensource.org/licenses/mit-license.php
+  Author Tobias Koppers @sokra
+*/
+import url from 'url';
+import async from 'async';
+import RawSource from 'webpack-sources/lib/RawSource';
+
+class CompressionPlugin {
+  constructor(options = {}) {
+    this.asset = options.asset || '[path].gz[query]';
+    this.algorithm = options.algorithm || 'gzip';
+    this.filename = options.filename || false;
+    this.compressionOptions = {};
+    if (typeof this.algorithm === 'string') {
+      if (this.algorithm === 'zopfli') {
+        try {
+          const zopfli = require('node-zopfli'); // eslint-disable-line no-unused-vars
+        } catch (err) {
+          throw new Error('node-zopfli not found');
+        }
+        this.compressionOptions = {
+          verbose: hasOwnProperty.call(options, 'verbose') ? options.verbose : false,
+          verbose_more: hasOwnProperty.call(options, 'verbose_more') ? options.verbose_more : false,
+          numiterations: options.numiterations ? options.numiterations : 15,
+          blocksplitting: hasOwnProperty.call(options, 'blocksplitting') ? options.blocksplitting : true,
+          blocksplittinglast: hasOwnProperty.call(options, 'blocksplittinglast') ? options.blocksplittinglast : false,
+          blocksplittingmax: options.blocksplittingmax ? options.blocksplittingmax : 15,
+        };
+        this.algorithm = (content, options, fn) => {
+          zopfli.gzip(content, options, fn);
+        };
+      } else {
+        const zlib = require('zlib');
+        this.algorithm = zlib[this.algorithm];
+        if (!this.algorithm) throw new Error('Algorithm not found in zlib');
```

Comment:
This is another case I would rather force by linting. `if (...) {}` always as it bites too easily.